### PR TITLE
Fix re-focusing problem related with usage of `Tab` (#885)

### PR DESCRIPTION
**Pull request by philiptzou**
This pull-request addressed the solution I proposed in #885.

### CODE REVIEW

File: src/datepicker.jsx [hunk 1]
```diff
@@ -259,7 +259,9 @@ export default class DatePicker extends React.Component {
 
   onInputKeyDown = (event) => {
     if (!this.state.open && !this.props.inline) {
-      this.onInputClick()
+      if (!(/^(Tab|Enter)$/.test(event.key))) {
```
Comment (rafeememon) [inline src/datepicker.jsx]: Can we avoid a regex here?  It adds some unnecessary obfuscation

Comment (alex-shamshurin) [inline src/datepicker.jsx]: I agree, no regex here, overhead.

Comment (philiptzou) [inline src/datepicker.jsx]: Well I guess it's true. I'll made the change.

**Comment by philiptzou** [comment]
Addressed the code review comment and also prevented to re-open the calendar when pressed "ESC" twice.

**Comment by cattermo** [comment]
Please release this code. This bug is stopping me from upgrading to React 15.6 :-(
@rafeememon @martijnrusschen

**Comment by martijnrusschen** [comment]
I’ll release this as soon as I get back from my vacation. Probably Monday.

**Pull request merged**